src/btree/bt_vrfy.c

```diff
@@ -1471,17 +1471,29 @@ __verify_page_discard(WT_SESSION_IMPL *session, WT_BM *bm)
      */
     __wt_qsort(page_ids, num_pages_found_in_btree, sizeof(uint64_t), __verify_compare_page_id);
 
-    for (size_t i = 0; i < num_pages_found_in_palm; i++) {
-        /* FIXME-WT-15451: Print mismatch page IDs for discard verify. */
-        if (((uint64_t *)item->data)[i] != page_ids[i]) {
-            /*
-             * FIXME-WT-14700: Investigate whether we need to do anything special when freeing a
-             * root page. Change below warning to an error after root page discard is implemented,
-             * if a mismatch is found this function will return the corresponding error code.
-             */
+    size_t i = 0, j = 0;
+    while (i <= num_pages_found_in_palm && j <= num_pages_found_in_btree) {
+        uint64_t id_in_palm = 0, id_in_btree = 0;
+        if (i == num_pages_found_in_palm && j == num_pages_found_in_btree) {
+            break;
+        }
```

Comment:
No curly brackets for one-liner:

```suggestion
        if (i == num_pages_found_in_palm && j == num_pages_found_in_btree)
            break;
```

Reply:
I'll use a better style here

Reply:
Fixed